data/lua/carryover_gold.lua

```diff
@@ -0,0 +1,248 @@
+local _ = wesnoth.textdomain "wesnoth"
+local carryover = {}
+
+--- Returns true iff there is not a single side that is controlled by the local human player.
+---@return boolean
+function carryover.get_is_observer()
+	for _, side in ipairs(wesnoth.sides) do
+		if side.controller == "human" and side.is_local then
+			return false
+		end
+	end
+	return true
+end
+
+--- Show names when thre are multiple persistent teams.
+---@return integer
```

Comment:
Please add a comment explaining the return value, like this:

`---@return integer #the number of things`